evaluate.py

```diff
@@ -0,0 +1,107 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+
+from pyro.contrib.forecast import backtest as pyro_backtest
+from pyro.ops.stats import crps_empirical, quantile
+
+from util import M5Data
+
+
+@torch.no_grad()
+def eval_mae(pred, truth):
+    """
+    Like pyro.contrib.forecast.eval_mae but does not average over batch dimensions.
+    """
+    pred = pred.median(0).values
+    return (pred - truth).abs().reshape(truth.shape[:-2] + (-1,)).mean(-1).cpu()
+
+
+@torch.no_grad()
+def eval_rmse(pred, truth):
+    """
+    Like pyro.contrib.forecast.eval_rmae but does not average over batch dimensions.
```

Comment:
we have weight and scale factors for each time series (among a batch), so we need to keep batch dimension for the result.

Reply:
nit: s/eval_rmae/eval_rmse

Reply:
@fritzo I am not sure what to do with the batch result of `mae`, `cprs`, so I'll use the Pyro version here instead.

Reply:
We should be able to aggregate `mae`, `crps` and `rmse` in exactly the same way, since they all are in units of data (# items sold). Can you explain your concern?

Reply:
So I believe for `mae` and `crps`, we will compute `scale` using `lag1_l1` norm instead of `l2` norm. Will do so.